edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/DiagnosticConfig.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Text;
+    using Microsoft.Extensions.Configuration;
+
+    public class DiagnosticConfig
+    {
+        public bool Enabled;
+        public string MetricsStoragePath;
+        public TimeSpan ScrapeInterval;
+        public TimeSpan UploadInterval;
+
+        public DiagnosticConfig(bool enabled, string storagePath, IConfiguration configuration)
+        {
+            this.Enabled = enabled;
+            this.MetricsStoragePath = Path.Combine("metrics", storagePath);
```

Comment:
if storage path is an absolute path, e.g. C:\test\abc, how Path.Combine handle this case? why not just add customer to define the complete storage path for metrics?

Reply:
well, it would have handled everything badly, since it should have been ```Path.Combine(storagePath, "metrics")```. I don't know how it worked when I tested it. should I require a new storage path, or keep it in this folder. The customer is never expected to interact with these files, these are the metrics temporarily stored until upload.